### PR TITLE
Fix logic bug in aggregator for skipping all classification results

### PR DESCRIPTION
**Pull request by psychiatrichwj**
In previous code change, we try to fetch tasks for a raw_data_id directly from redis, and in case of err, skip the job. This can be wrong since in case of the key not exists, it will always be error, so fix this by checking exists or not first and then try to fetch the tasks.

We also removed the logic to remove the key from redis, so we can still keep it there for reference later

### CODE REVIEW

File: nostr-client-plus/src/bin/aggregator.rs [hunk 1]
```diff
@@ -306,28 +306,40 @@ async fn run() -> Result<()> {
...
+                    };
                 }
                 Err(err) => {
-                    tracing::error!("Failed to fetch tasks from redis for data: {}, error: {}", raw_data_id.clone(), err);
+                    tracing::error!("Failed to check if tasks exits from redis for data: {}, error: {}", raw_data_id.clone(), err);
                     // Skip this task since something wrong with redis
                     continue;
                 }
-            };
+                _ => {}
```
Comment (germanop) [inline nostr-client-plus/src/bin/aggregator.rs]: No need to use the "catch-all" branch, there is only another case you did not cover, and it's `Ok(false)` so, please, be explicit.

That said, I do not understand why we need to `match` twice: if we just need to check for existence, then just the first match, but if you need to retrieve the key, then just call `get` and then handle the result.
If there is already a vector, good. If there is nothing, then create an empty vec and assign to `tasks`. If there is an error, do what you already did.

So, bottom line is, do not check for existence and then retrieve, just retrieve.

I could not test the `get` method because I have no redis, but because it's just a wrapper for Redis' `GET` and `MGET`, I would expect the same behaviour.

File: nostr-client-plus/src/bin/aggregator.rs [hunk 2]
```diff
@@ -306,28 +306,40 @@ async fn run() -> Result<()> {
...
+                    tasks = match saved_tasks {
                         Ok(val) => {
-                            val
-                        }
+                            match from_str(val.as_str()) {
+                                Ok(val) => {
+                                    val
+                                }
 
+                                Err(err) => {
```
Comment (germanop) [inline nostr-client-plus/src/bin/aggregator.rs:319]: No blank lines between match branches

**Review by germanop — CHANGES_REQUESTED** [review]

File: nostr-client-plus/src/bin/aggregator.rs (hunk 1 above)
Comment (psychiatrichwj) [reply nostr-client-plus/src/bin/aggregator.rs]: In case the key not exists, it returns an error, so just want to be explicit since distinguishing between errors can be annoying.

**Review by germanop — APPROVED** [review]
